community/detectors/joomla_cve_2023_23752/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202323752/Cve202323752VulnDetector.java

```diff
@@ -0,0 +1,158 @@
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("CVE_2023_23752"))
+                .setSeverity(Severity.HIGH)
+                .setTitle("Joomla unauthorized access to webservice endpoints")
+                .setDescription(
+                    "CVE-2023-23752: An improper access check allows unauthorized access to"
+                        + " webservice endpoints. attacker can get the host address "
+                        + "and username and password of the configured joomla database.")
+                .setRecommendation("Upgrade Joomla to 4.2.8 and above versions.")
+                .addAdditionalDetails(
+                    AdditionalDetail.newBuilder()
+                        .setTextData(TextData.newBuilder().setText(exposedConfig))))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    HttpHeaders httpHeaders =
+        HttpHeaders.builder()
+            .addHeader(
+                ACCEPT,
+                "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8")
+            .build();
+
+    String appConfUrl =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService)
+            + "api/index.php/v1/config/application?public=true";
+    try {
+      HttpResponse appConfHttpResponse =
+          httpClient.send(get(appConfUrl).setHeaders(httpHeaders).build(), networkService);
+
+      // immediate checks for accelerating the scan
+      if (appConfHttpResponse.status().code() != HttpStatus.OK.code()
+          || appConfHttpResponse.bodyJson().isEmpty()
+          || appConfHttpResponse.bodyString().isEmpty()) {
+        return false;
+      }
+
+      // check for body values match our detection rules
+      if (appConfHttpResponse.bodyString().get().contains("password")
+          && appConfHttpResponse.bodyString().get().contains("user")) {
+
+        JsonObject jsonResponse = (JsonObject) appConfHttpResponse.bodyJson().get();
+        if (jsonResponse.keySet().contains("data")) {
+          JsonArray jsonArray = jsonResponse.getAsJsonArray("data");
+          for (int i = 0; i < jsonArray.size(); i++) {
+            if (jsonArray.get(i).getAsJsonObject().keySet().contains("attributes")) {
+              exposedConfig = appConfHttpResponse.bodyString().get();
+              return true;
+            }
+          }
+        }
+      }
+    } catch (IllegalStateException | JsonSyntaxException | IOException | AssertionError e) {
```

Comment:
```suggestion
    } catch (NoSuchElementException | IllegalStateException | JsonSyntaxException e) {
      return false;
    } catch (IOException e) {
```
JSON errors should be catched in a different block so we can return silently in case the target we're scanning is not actually a Joomla instance or is patched.